### DISCUSSION TITLE
Re-include support for openSUSE Tumbleweed (rolling release version)

### DISCUSSION THREAD

**Discussion by cyril279** [Ulauncher/Ulauncher#884] [Packaging]
## Support openSUSE Tumbleweed!
Support for openSUSE was dropped at `5.0.0.r5 (Dev)` due to python3 migration, which requires python 3.5+.

**openSUSE Tumbleweed (Rolling release) uses python 3.7**

With the addition of `typelib-1_0-Keybinder-3_0` and `python3-pyinotify` packages, `ulauncher_5.3.0.beta1_fedora29.rpm` installs and runs happily on Tumbleweed.

Can we have Tumbleweed rpm builds?

Great app.

```
human@linux-zabz:~> zypper if python3

Information for package python3:
--------------------------------
Repository     : Main Repository (OSS)
Name           : python3              
Version        : 3.7.2-3.2            
Arch           : x86_64               
Vendor         : openSUSE             
Installed Size : 165.5 KiB            
Installed      : Yes (automatically)  
Status         : up-to-date           
Source package : python3-3.7.2-3.2.src
Summary        : Python 3 Interpreter 

             .;ldkO0000Okdl;.                human@linux-zabz
         .;d00xl:^''''''^:ok00d;.            OS: openSUSE 20190607
       .d00l'                'o00d.          Kernel: x86_64 Linux 5.1.7-1-default
     .d0Kd'  Okxol:;,.          :O0d.        Uptime: 22m
    .OKKKK0kOKKKKKKKKKKOxo:,      lKO.       Packages: 2292
   ,0KKKKKKKKKKKKKKKK0P^,,,^dx:    ;00,      Shell: bash 5.0.7
  .OKKKKKKKKKKKKKKKKk'.oOPPb.'0k.   cKO.     Resolution: 1920x1200
  :KKKKKKKKKKKKKKKKK: kKx..dd lKd   'OK:     DE: XFCE4
  dKKKKKKKKKKKOx0KKKd ^0KKKO' kKKc   dKd     WM: Mutter
  dKKKKKKKKKKKK;.;oOKx,..^..;kKKK0.  dKd     WM Theme: 
  :KKKKKKKKKKKK0o;...^cdxxOK0O/^^'  .0K:     GTK Theme: Adwaita [GTK2]
   kKKKKKKKKKKKKKKK0x;,,......,;od  lKk      Icon Theme: Adwaita
   '0KKKKKKKKKKKKKKKKKKKKK00KKOo^  c00'      Font: Noto Sans 11
    'kKKKOxddxkOO00000Okxoc;''   .dKk'       CPU: Intel Xeon E5640 @ 8x 2.661GHz [46.0°C]
      l0Ko.                    .c00l'        GPU: NV94
       'l0Kk:.              .;xK0l'          RAM: 1333MiB / 11997MiB
          'lkK0xl:;,,,,;:ldO0kl'            
              '^:ldxkkkkxdl:^'              

```

**Comment by gornostal** [comment]
Yes, I'll put this on top of my todo list.

**Comment by gornostal** [comment]
I got stuck with getting this line to work:
```
>>> from gi.repository import Gdk
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.7/site-packages/gi/importer.py", line 133, in load_module
    'introspection typelib not found' % namespace)
ImportError: cannot import name Gdk, introspection typelib not found
```
🤔

Installed all the necessary dependencies, or so I think.
I'm testing it from a docker container with the latest tumbleweed version.

Any help is appreciated.

**Comment by luismanson** [comment]
why not provide packages trough opensuse OBS? You can build for other distros too.
Im stuck at:

`Traceback (most recent call last):
  File "/usr/bin/ulauncher", line 27, in <module>
    from ulauncher.main import main
ModuleNotFoundError: No module named 'ulauncher'`

**Comment by Slickytail** [comment]
> I got stuck with getting this line to work:
> 
> ```
> >>> from gi.repository import Gdk
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/usr/lib64/python3.7/site-packages/gi/importer.py", line 133, in load_module
>     'introspection typelib not found' % namespace)
> ImportError: cannot import name Gdk, introspection typelib not found
> ```
> 
> thinking
> 
> Installed all the necessary dependencies, or so I think.
> I'm testing it from a docker container with the latest tumbleweed version.
> 
> Any help is appreciated.

I have not been able to recreate this. I had no issues getting it to build in whatever the default container is (whatever container gets picked by `ul dev-container`)

**Comment by LorenDB** [comment]
I'd like to bump this issue by saying that I use openSUSE Tumbleweed on my work machine, and literally the only thing left keeping me from switching to Tumbleweed on my personal laptop is the lack of working Ulauncher. I'd really appreciate if an openSUSE release could be made.

Thanks for the great app. As I mentioned, I use it all the time. It even beats out KRunner, not because it has more functionality built in, but because it looks and feels better. Keep up the good work!

(Edit: 6 months later, I am happy with KRunner.)
